tests/__utils__/mocks.ts

```diff
@@ -18,10 +18,10 @@ export function spyOnActions(): jest.Mocked<SearchActions> {
 
 export type RecursivePartial<T> = {
   [P in keyof T]?:
-  T[P] extends (infer U)[] ? RecursivePartial<U>[] :
-    // eslint-disable-next-line @typescript-eslint/ban-types
-    T[P] extends object ? RecursivePartial<T[P]> :
-      T[P];
+  T[P] extends undefined ? undefined :
```

Comment:
I looked around a little and it seems like our original version of RecursivePartial was taken from a StackOverflow [post](https://stackoverflow.com/questions/41980195/recursive-partialt-in-typescript). A few other places suggest this as well, because there's no TS utility that currently does this. I tried looking for a 3rd party library, but the couple I was able to find (e.g. `DeepPartial` from https://github.com/joonhocho/tsdef) don't quite work for our purposes because they make primitive types partials too.

A comment in the original StackOverflow post I linked above actually suggests almost the same change I made because they encountered the same problem with optional array and object types not working correctly